api/envoy/config/route/v3/route_components.proto

```diff
@@ -2168,6 +2169,27 @@ message RateLimit {
     }
   }
 
+  // [#not-implemented-hide:]
+  message HitsAddend {
+    // Fixed number of hits to add to the rate limit descriptor. Only one of the ``number`` or
+    // ``format`` fields can be set.
+    google.protobuf.UInt32Value number = 1 [(validate.rules).uint32 = {lte: 100000000}];
+
+    // Substitution format string to extract the number of hits to add to the rate limit descriptor.
+    // The same :ref:`format specifier <config_access_log_format>` as used for
+    // :ref:`HTTP access logging <config_access_log>` applies here.
+    //
+    // .. note::
+    //   The format string must contain only single valid substitution field that will be replaced
+    //   with a non-negative number.
+    //
+    // For example, the ``%BYTES_RECEIVED%`` format string will be replaced with the number of bytes
+    // received in the request.
+    //
+    // Only one of the ``number`` or ``format`` fields can be set.
```

Comment:
what will happen if both are set?

Reply:
This is oneof semantics. (although oneof is not recommend according to our style).

If both are set, the configuration will be rejected.